tests/tests/class-linking.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace HM\Tachyon\Tests;
+
+use Tachyon;
+use WP_UnitTestCase;
+
+/**
+ * Test rescheduling an event is successful.
+ *
+ * @ticket 64
+ */
+class Tests_Linking extends WP_UnitTestCase {
+
+	/**
+	 * @var int[] Attachment IDs
+	 */
+	static $attachment_ids;
+
+	/**
+	 * Set up attachments and posts require for testing.
+	 *
+	 * tachyon.jpg: 1280x719
+	 * tachyon-large.jpg: 5312x2988
+	 * Photo by Digital Buggu from Pexels
+	 * @link https://www.pexels.com/photo/0-7-rpm-171195/
+	 */
+	static public function wpSetUpBeforeClass( $factory ) {
+		/**
+		 * Filter the uploads directory to avoid file name clashes in
+		 * subsequent tests.
+		 */
+		add_filter( 'upload_dir', function( $upload_dir ) {
+			$dir = strtolower( str_replace( [ '_', '/', '\\' ], '-', __CLASS__ ) );
+			$upload_dir['basedir'] = preg_replace( '#/uploads(/|$)#', "/uploads/{$dir}\$1", $upload_dir['basedir'] );
+			$upload_dir['path'] = preg_replace( '#/uploads(/|$)#', "/uploads/{$dir}\$1", $upload_dir['path'] );
+			return $upload_dir;
+		} );
+
+		self::$attachment_ids['tachyon'] = $factory->attachment->create_upload_object(
+			realpath( __DIR__ . '/../data/tachyon.jpg')
+		);
+	}
+
+	/**
+	 * Runs the routine after all tests have been run.
+	 */
+	public static function tearDownAfterClass() {
+		array_map( function( $post_id ) {
+			wp_delete_attachment( $post_id, true );
+		}, self::$attachment_ids );
+	}
+
+	/**
+	 * Extract the first src attribute from the given HTML.
+	 *
+	 * There should only ever be one image in the content so regex
+	 * can be dropped in favour of strpos techniques for getting the
+	 * src of an image.
+	 *
+	 * @param $html string HTML containing an image tag.
+	 * @return string The first `src` attribute within the first image tag.
+	 */
+	function get_src_from_html( $html ) {
```

Comment:
I feel like there should be a simpler way to do this with a regex?

Reply:
Regex used for src and href extraction in 96da9037403bcfe5ac31187f2a9ea1a054a25578.